pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -340,6 +330,20 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 		// If the ordinal could not be parsed (ord < 0), ignore the Pod.
 	}
 
+	// make sure to update the latest status even if there is an error later
+	defer func() {
+		// update the set's status
+		updateErr := ssc.updateStatefulSetStatus(ctx, set, &status)
+		if updateErr != nil && err == nil {
+			err = updateErr
```

Comment:
I'm not sure this variable `err` is capturing all the errors.

For example, line 428 instantiates a new variable.

I think the best way to capture the returning error is to name the return values in line 267.

Reply:
Are you sure this is the right variable? Line 269 might be covering it up.

Reply:
Hmm...I'm not really sure. It's ok in my tests. But it's definitely better to name it  another value.  I've change the code.